script/20240516-revoke-roles/02_GrantAdminRoleForMultisig_Mainnet.s.sol

```diff
@@ -0,0 +1,362 @@
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { AccessControlEnumerable } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import { LibRNSDomain } from "src/libraries/LibRNSDomain.sol";
+import { DefaultNetwork } from "@fdk/utils/DefaultNetwork.sol";
+import { RNSUnified } from "src/RNSUnified.sol";
+import { RNSAuction } from "src/RNSAuction.sol";
+import { NameChecker } from "src/NameChecker.sol";
+import { RNSDomainPrice } from "src/RNSDomainPrice.sol";
+import { PublicResolver } from "src/resolvers/PublicResolver.sol";
+import { RNSReverseRegistrar } from "src/RNSReverseRegistrar.sol";
+import { RONRegistrarController } from "src/RONRegistrarController.sol";
+import { OwnedMulticaller } from "src/utils/OwnedMulticaller.sol";
+import { INSDomainPrice } from "src/interfaces/INSDomainPrice.sol";
+import { OwnedMulticallerDeploy } from "script/contracts/OwnedMulticallerDeploy.s.sol";
+import { ErrorHandler } from "src/libraries/ErrorHandler.sol";
+import { EventRange } from "src/libraries/LibEventRange.sol";
+import { RNSOperation } from "src/utils/RNSOperation.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { INSAuction } from "src/interfaces/INSAuction.sol";
+import { INSDomainPrice } from "src/interfaces/INSDomainPrice.sol";
+
+contract Migration__02_GrantAdminRoleForMultisig_Mainnet is Migration {
+  using Strings for *;
+  using ErrorHandler for bool;
+  using LibRNSDomain for string;
+
+  address duke = 0x0F68eDBE14C8f68481771016d7E2871d6a35DE11;
+  address multisig = 0x9d05d1f5b0424f8fde534bc196ffb6dd211d902a;
+
+  RNSUnified internal _rns;
+  RNSAuction internal _auction;
+  NameChecker internal _nameChecker;
+  RNSDomainPrice internal _domainPrice;
+  PublicResolver internal _publicResolver;
+  RNSReverseRegistrar internal _reverseRegistrar;
+  RONRegistrarController internal _ronController;
+  OwnedMulticaller internal _ownedMulticaller;
+  RNSOperation internal rnsOperation;
+  address internal _batchTransfer;
+
+  function run() external onlyOn(DefaultNetwork.RoninMainnet.key()) {
+    _rns = RNSUnified(loadContract(Contract.RNSUnified.key()));
+    _auction = RNSAuction(loadContract(Contract.RNSAuction.key()));
+    _nameChecker = NameChecker(loadContract(Contract.NameChecker.key()));
+    _domainPrice = RNSDomainPrice(loadContract(Contract.RNSDomainPrice.key()));
+    _publicResolver = PublicResolver(loadContract(Contract.PublicResolver.key()));
+    _reverseRegistrar = RNSReverseRegistrar(loadContract(Contract.RNSReverseRegistrar.key()));
+    _ronController = RONRegistrarController(loadContract(Contract.RONRegistrarController.key()));
+    // Verify: https://app.roninchain.com/address/0x27876429DB2cDDF017DBb63560D0366E4B4E6f8a
+    _ownedMulticaller = OwnedMulticaller(0x27876429DB2cDDF017DBb63560D0366E4B4E6f8a);
```

Comment:
Current owner of multicaller is Tu. Please transfer the ownership to appropriate address.

Reply:
Tu will transfer his ownership to multisig once system works as expected after revoke operation